src/hwb.rs

```diff
@@ -0,0 +1,287 @@
+    ///It varies from 0 to 1, with 1 being always full black and 0 always
+    ///being the color tint (a mixture of a pure hue with white) chosen with the other two
+    //controls.
+    pub blackness: T,
+}
+
+impl<T: Float> Hwb<T> {
+    ///Linear HWB.
+    pub fn new(hue: RgbHue<T>, whiteness: T, blackness: T) -> Hwb<T> {
+        Hwb {
+            hue: hue,
+            whiteness: whiteness,
+            blackness: blackness,
+        }
+    }
+}
+
+///<span id="Hwba"></span>[`Hwba`](type.Hwba.html) implementations.
+impl<T: Float> Alpha<Hwb<T>, T> {
+    ///Linear HSV and transparency.
+    pub fn new(hue: RgbHue<T>, whiteness: T, blackness: T, alpha: T) -> Hwba<T> {
+        Alpha {
+            color: Hwb::new(hue, whiteness, blackness),
+            alpha: alpha,
+        }
+    }
+}
+
+impl<T: Float> FromColor<T> for Hwb<T> {
+    fn from_xyz(xyz: Xyz<T>) -> Self {
+        let hsv: Hsv<T> = Hsv::from_xyz(xyz);
+        Self::from_hsv(hsv)
+    }
+
+    fn from_hsv(hsv: Hsv<T>) -> Self {
+        Hwb {
+            hue: hsv.hue,
+            whiteness: (T::one() - hsv.saturation) * hsv.value,
+            blackness: (T::one() - hsv.value),
+        }
+    }
+
+    fn from_hwb(hwb: Hwb<T>) -> Self {
+        hwb
+    }
+
+}
+
+impl<T: Float> Limited for Hwb<T> {
+    fn is_valid(&self) -> bool {
+        self.blackness >= T::zero() && self.blackness <= T::one() &&
+        self.whiteness >= T::zero() && self.whiteness <= T::one()
+    }
+
+    fn clamp(&self) -> Hwb<T> {
+        let mut c = *self;
+        c.clamp_self();
+        c
+    }
+
+    fn clamp_self(&mut self) {
+        self.whiteness = clamp(self.whiteness, T::zero(), T::infinity());
+        self.blackness = clamp(self.blackness, T::zero(), T::infinity());
+        let sum = self.blackness + self.whiteness;
+        if sum > T::one() {
+            self.whiteness = self.whiteness / sum;
+            self.blackness = self.blackness / sum;
+        }
+    }
+}
+
+impl<T: Float> Mix for Hwb<T> {
+    type Scalar = T;
+
+    fn mix(&self, other: &Hwb<T>, factor: T) -> Hwb<T> {
+        let factor = clamp(factor, T::zero(), T::one());
+        let hue_diff: T = (other.hue - self.hue).to_degrees();
+
+        Hwb {
+            hue: self.hue + factor * hue_diff,
+            whiteness: self.whiteness + factor * (other.whiteness - self.whiteness),
+            blackness: self.blackness + factor * (other.blackness - self.blackness),
+        }
+    }
+}
+
+impl<T: Float> Shade for Hwb<T> {
+    type Scalar = T;
+
+    fn lighten(&self, amount: T) -> Hwb<T> {
+        Hwb {
+            hue: self.hue,
+            whiteness: self.whiteness + amount,
+            blackness: self.blackness - amount,
+        }
+    }
+}
+
+impl<T: Float> GetHue for Hwb<T> {
+    type Hue = RgbHue<T>;
+
+    fn get_hue(&self) -> Option<RgbHue<T>> {
+        if self.whiteness == T::zero() || self.blackness == T::zero() {
```

Comment:
Wait, what? I'm talking about `get_hue`, not `lighten`. Returning `None` if `self.whiteness == T::zero() || self.blackness == T::zero()` is not right at all. That would make it return `None` along the left and upper edges of those color swatches. It should rather check for `self.whiteness + self.blackness >= 1.0`.